### ISSUE TITLE
Metric's in Fiji

### ISSUE THREAD

**Issue by Maik-Lu** [embl-cba/fiji-plugin-elastixWrapper#1]
I'm using Elastix for 2D/3D registrations. Since I'm using Fiji for most of my image processing steps I was quite happy to find this plugin.  I have a short question: Since I'm normaly using "MultiMetricMultiResolutionRegistration" with Mattes Mutal and Euclidean Distance Metric, I was wondering if it's possible to change the metric within the Fiji plugin. Normally I search for Landmarks in Fiji and than do the registration in Elastix and it would be really straight forward to do everything in Fiji.

**Comment by tischi** [comment]
Thank you very much for your interest!

For multi-channel images, I am already using `MultiMetricMultiResolutionRegistration`. 
- https://github.com/embl-cba/fiji-plugin-elastixWrapper/blob/master/src/main/java/de/embl/cba/elastixwrapper/elastix/ElastixParameters.java#L91

And in general, I am using  `AdvancedMattesMutualInformation` for the metric. 
- hhttps://github.com/embl-cba/fiji-plugin-elastixWrapper/blob/master/src/main/java/de/embl/cba/elastixwrapper/elastix/ElastixParameters.java#L139

I also had a look at the `CorrespondingPointsEuclideanDistanceMetric` and it seems that one needs to provide additional landmark information in the form of two text files. Do you have experience with this? I never used this... But it could be very interesting, because I typically have the problem that images need to be quite well pre-aligned for elastix not to get stuck in local minima. Is that where the landmarks help?

If you you are interested in more manual landmark based registration this might be also interesting for you: https://imagej.net/BigWarp

**Comment by Maik-Lu** [comment]
Thank's for the answer. Yes, I alway use two text files for the Landmarks for CPEuclideanDistancesMetric in addition to  AdvancedMattesMutualInformation. Without the Points in 3D/2D registration it always stucks somewhere/doesn't work at all. I think it'll help you because it works for me without any pre-alignment.

**Comment by tischi** [comment]
@Maik-Lu,  That sounds super interesting!
And how do you generate the matching landmark points in Fiji? 
Did you write a macro or plugin for that?

**Comment by Maik-Lu** [comment]
@tischi, yes this is eher the problem starts. I open the images in Fiji and look for at least three points which I find in both images and write them down manualy in a txt file. Then I use the points for Elastixs in ITK.

**Comment by tischi** [comment]
@Maik-Lu 
Sounds like the workflow could be improved :-)
Did you ever try https://imagej.net/BigWarp?

**Comment by Maik-Lu** [comment]
@tischi 
No, I I didn't, the nice thing about Elastix is that I can use both metrics at the same time and define different weights for the two. The points help that the 3D image remains around the position of the 2D section / local minima are avoided, but the actual fine-tuning is done by AdvancedMattesMutualInformation.

**Comment by tischi** [comment]
@Maik-Lu 
OK! I think adding a user interface for convenient selection of pairs of points would be super nice, and I will definitely do it at some point, but I am a bit overloaded with other work at the minute... 
Could you maybe be me a favor and try out whether the way this is done in BigWarp feels good to you? Because, in fact, my idea would be to steal their code....

**Comment by Maik-Lu** [comment]
@tischi nice - looking forward to it. Next week I will have some new VisNIR images, than I will definitely play around with BigWarp.

**Comment by Maik-Lu** [comment]
@tischi 
It's a really nice plugin and easy to use - I will now also use it to safe the landmarks for Elastix. A combination of your plugin with AdvancedMattesMutualInformation and the addition of the point selection like in BigWarp for EuclideanDistanceMetric would be a super plugin for 2D/3D or 3D/3D registration.

**Comment by tischi** [comment]
Very good to hear!
Maybe a first step could be that I read within my plugin directly the landmark files from BigWarp?
Could you provide me with such a file as written by BigWarp? 
Ideally together with some actual image data, such that I can test it.

**Comment by Maik-Lu** [comment]
@tischi 
Cool - I sent you small images as examples. 
I often use Pyramideshedule to enhance the performance, Elastix is much faster with it. Do you plan to implement this in the plugin too?
